pls_core/src/inflections/mod.rs

```diff
@@ -106,3 +106,82 @@ where
         Ok(result)
     }
 }
+
+fn get_inflections_for_pattern(
+    pattern: &str,
+    exec_sql: impl Fn(String) -> Result<Vec<Vec<Vec<String>>>, String>,
+) -> Result<Vec<Vec<Vec<String>>>, String> {
+    exec_sql("Select * from {}".replace("{}", pattern))
+}
+
+fn get_words_for_indeclinable_stem(paliword: &str) -> Vec<(String, String, String, String)> {
+    vec![(
+        paliword.chars().filter(|c| !c.is_digit(10)).collect(),
+        paliword.to_string(),
+        " ".to_string(),
+        "ind".to_string(),
+    )]
+}
+
+fn get_words_for_irregular_stem(
+    paliword: &str,
+    pattern: &str,
+    _exec_sql: fn(&str) -> Result<String, String>,
+) -> Vec<(String, String, String, String)> {
+    let inflections: Vec<Vec<String>> =
+        get_inflections_for_pattern(pattern, exec_sql_structured(_exec_sql))
+            .unwrap()
+            .pop()
+            .unwrap();
+    let mut inflected_words_irregular_stem: Vec<(String, String, String, String)> = Vec::new();
+    for mut inflection_row in inflections {
+        for inflection in inflection_row.pop().unwrap().split(',') {
+            inflected_words_irregular_stem.push((
+                inflection.to_string(),
+                paliword.to_string(),
+                inflection_row.join(" ").to_string(),
+                "*".to_string(),
+            ))
+        }
+    }
+    inflected_words_irregular_stem
+}
+
+fn get_words_for_regular_stem(
+    paliword: &str,
+    stem: &str,
+    pattern: &str,
+    _exec_sql: fn(&str) -> Result<String, String>,
+) -> Vec<(String, String, String, String)> {
```

Comment:
optional comment - this should be returning as Result<Vec<(...)>

using unwrap is bad as it cause random panics from middle of the code, instead of the error conditions nicely bubbling up to the top where they can be handled appropriately (passed to use, logged, ignored etc.).

its optional - so you can push it through without fixing.